background.js

```diff
@@ -133,79 +90,66 @@ function modifyReferer(e)
 
 
 /*
- * This function is called when the inject-referer.js content script
- * asks for the value to set as document.referrer.
+ * Listener function for storage changes: Forward any changes to the
+ * global variables.
  */
-function handleContentMsg(request, sender, sendResponse) {
-	const conf = findHostConf(request.target, request.referrer);
-	var referrer = null;
-	switch (conf.action)
-	{
-		case "prune":
-			referrer = new URL(request.referrer).origin + "/";
-			break;
-		case "target":
-			referrer = new URL(request.target).origin + "/";
-			break;
-		case "replace":
-			referrer = conf.referer;
-			break;
-		case "remove":
-			referrer = '';
-			break;
-		default:
-			/* "keep" */
-			referrer = request.referrer;
+async function refreshConfig(change, area)
+{
+	let dirty = false;
+	if (area === "sync") {
+		if (change.hasOwnProperty("domains")) {
+			config.domains = change.domains.newValue;
+			dirty = true;
+		}
+		if (change.hasOwnProperty("any")) {
+			config.anyConf = change.any.newValue;
+			dirty = true;
+		}
+		if (change.hasOwnProperty("same")) {
+			config.sameConf = change.same.newValue;
+			dirty = true;
+		}
 	}
-	sendResponse({referrer: referrer});
-}
 
-
-
-/*
- * Escape function from
- * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
- */
-function escapeRegExp(string)
-{
-	return string.replace(/[.*+?^${}()|\[\]\\]/g, '\\$&');
-	// $& means the whole matched string
+	if (dirty) {
```

Comment:
```suggestion
	if (isDirty) {
```

Reply:
Let's wait for airtower-luna's opinion on coding styles.

Reply:
I prefer single-word variable names, as long as they're sufficiently descriptive. `dirty` fits that, `changed` or something likes that would be better.

Reply:
I think `dirty` as in a "dirty bit".

Reply:
Like for a cache, yes, but in this case unnecessarily generic. Just a style issue though. :slightly_smiling_face: